comp/dogstatsd/server/server.go

```diff
@@ -630,6 +630,9 @@ func (s *server) parsePackets(batcher *batcher, parser *parser, packets []*packe
 				}
 			}
 		}
+		packet.Buffer = nil
+		packet.Contents = nil
+		packet.Origin = ""
```

Comment:
Ah, I see. This buffer is what's being pooled here. I misunderstood and thought there was another buffer in play.